src/utils/service_utils.ts

```diff
@@ -106,7 +106,12 @@ export const serviceUtils = {
         }, []);
     },
     getAffiliateFeeAmounts(quote: SwapQuote, fee: AffiliateFee): AffiliateFeeAmounts {
-        if (fee.feeType === AffiliateFeeType.None || fee.recipient === NULL_ADDRESS || fee.recipient === '') {
+        if (
+            fee.feeType === AffiliateFeeType.None ||
+            fee.recipient === NULL_ADDRESS ||
+            fee.recipient === '' ||
+            (fee.feeType === AffiliateFeeType.PositiveSlippageFee && quote.orders.length === 1)
```

Comment:
What's the intention here where quote `orders.length === 1`

Reply:
I don't want to add `affiliateFee` gas overhead if the quote has `PositiveSlippageFee` and if we're only routing to 1 order (VIP). Just kind of like here: https://github.com/0xProject/protocol/blob/6d877d52425230ba7664a0685ab0a06957108acb/packages/asset-swapper/src/quote_consumers/quote_consumer_utils.ts#L82

Reply:
What do you think?